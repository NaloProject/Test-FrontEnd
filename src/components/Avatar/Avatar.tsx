import { FC } from 'react'
import Image from 'next/image'
import clsx from 'clsx'

export type AvatarProps = {
  alt: string
  className?: string
  size: number
  src: string
}

const Avatar: FC<AvatarProps> = ({ alt, className, size, src }) => (
  <div
    className={clsx(
      `overflow-hidden rounded-full h-[${size}px] w-[${size}px]`,
      className
    )}
  >
    <Image src={src} width={size} height={size} alt={alt} />
  </div>
)

export default Avatar
