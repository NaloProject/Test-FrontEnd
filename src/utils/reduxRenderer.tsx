import { configureStore, EnhancedStore } from '@reduxjs/toolkit'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ReactElement, FC, PropsWithChildren } from 'react'
import { Provider } from 'react-redux'

import { combined, RootState } from '@Redux/store'
import INITIAL_STATE from './initialState'

type RenderWithReduxOptions = Omit<RenderOptions, 'queries'> & {
  initialState?: RootState
  store?: EnhancedStore
}

export const renderWithRedux = (
  ui: ReactElement,
  {
    initialState = INITIAL_STATE,
    store = configureStore({
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
          immutableCheck: false,
          serializableCheck: false,
        }),
      preloadedState: initialState,
      reducer: combined,
    }),
    ...renderOptions
  }: RenderWithReduxOptions = {}
): RenderResult => {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => (
    <Provider store={store}>{children}</Provider>
  )

  return render(ui, { wrapper: Wrapper, ...renderOptions })
}
