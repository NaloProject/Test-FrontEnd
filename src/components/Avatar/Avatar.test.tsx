import { renderWithRedux } from '@Utils/reduxRenderer'
import Avatar from './Avatar'

const IMG_URL =
  'https://d33wubrfki0l68.cloudfront.net/0834d0215db51e91525a25acf97433051f280f2f/c30f5/img/redux.svg'

describe('Avatar', () => {
  it('should mount first', async () => {
    const { container } = renderWithRedux(
      <Avatar alt="test" size={32} src={IMG_URL} />
    )

    expect(container).toMatchSnapshot()
  })
})
