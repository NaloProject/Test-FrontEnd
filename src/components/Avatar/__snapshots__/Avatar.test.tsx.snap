// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should mount first 1`] = `
<div>
  <div
    class="overflow-hidden rounded-full h-[32px] w-[32px]"
  >
    <img
      alt="test"
      data-nimg="1"
      decoding="async"
      height="32"
      loading="lazy"
      src="https://d33wubrfki0l68.cloudfront.net/0834d0215db51e91525a25acf97433051f280f2f/c30f5/img/redux.svg"
      style="color: transparent;"
      width="32"
    />
  </div>
</div>
`;
